src/pgen/cluster/agn_triggering.cpp

```diff
@@ -485,15 +522,36 @@ parthenon::TaskStatus
 AGNTriggeringFinalizeTriggering(parthenon::MeshData<parthenon::Real> *md,
                                 const parthenon::SimTime &tm) {
   auto hydro_pkg = md->GetBlockData(0)->GetBlockPointer()->packages.Get("Hydro");
-  const auto &agn_triggering = hydro_pkg->Param<AGNTriggering>("agn_triggering");
+  auto &agn_triggering = hydro_pkg->Param<AGNTriggering>("agn_triggering");
+  parthenon::Real cold_mass = 0.0; 
+  parthenon::Real total_mass = 0.0; 
+  parthenon::Real inflow_cold = 0.0; 
+  parthenon::Real inflow_tot = 0.0; 
+  if (agn_triggering.triggering_mode_ == AGNTriggeringMode::COLD_GAS) {
+    cold_mass = hydro_pkg->Param<Real>("agn_triggering_cold_mass"); 
+    total_mass = hydro_pkg->Param<Real>("agn_triggering_total_mass"); 
+
+
+    inflow_cold = (cold_mass - hydro_pkg->Param<Real>("agn_triggering_prev_cold_mass")) / tm.dt; 
+    inflow_tot = (total_mass - hydro_pkg->Param<Real>("agn_triggering_prev_total_mass")) / tm.dt; 
+
+  }
+  hydro_pkg->UpdateParam<AGNTriggering>("agn_triggering", agn_triggering); 
```

Comment:
This line can probably go as it's not required (`agn_triggering` is not modified in the code above) and might raise an error because you're updating a non-mutable object.